portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,65 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private DatastoreService datastore;
+
+  @Override
+  public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  private String convertToJsonUsingGson(ArrayList<Comment> comments) {
+    String json = new Gson().toJson(comments);
+    return json;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json;");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String user = (String) entity.getProperty("user");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String content = (String) entity.getProperty("content");
+
+      Comment comment = new Comment(id, user, timestamp, content);
+      comments.add(comment);
+    }
+    String ret = convertToJsonUsingGson(comments);
+    response.getWriter().println(ret);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+    String user = request.getParameter("user");
```

Comment:
Before create commentEntity, maybe think about what content is valid for this parameter.
for example null or "" .etc